File: 11729-hanoi.cc
#include <iostream>
#include <vector>
#include <utility>

std::vector<std::pair<char, char>> output;

int hanoi(int n, char _from, char _to, char _for) {
    if (n == 0) return 1;
    int sum {0};

    sum += hanoi(n-1, _from, _for, _to);
    output.push_back(std::pair<char, char>(_from, _to));
    sum += hanoi(n-1, _for, _to, _from);

    return sum;
}

int main() {
    int N;
    std::cin >> N;

    int count = hanoi(N, '1', '3', '2') - 1;
    std::cout << count << '\n';

    for (std::pair<char, char>& p : output) {
        std::cout << p.first << ' ' << p.second << '\n';
    }
}

File: 11729-hanoi.py
n = int(input())

output = []
def f(n, _from, _to, _for):
        if n is 0: return 1
        s=0
        s += f(n-1, _from, _for, _to)
        output.append(f"{_from} {_to}")
        s += f(n-1, _for, _to, _from)
        return s

print(f(n, '1', '3', '2') - 1)
print("\n".join(output))